frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,15 +1,79 @@
-type Properties = {
+import { type ReactNode } from 'react';
+
+import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
+
+const ButtonVariant = {
+    PRIMARY: 'primary',
+    SECONDARY: 'secondary',
+    TERTIARY: 'tertiary',
+} as const;
+
+const ButtonSize = {
+    SMALL: 'small',
+    MEDIUM: 'medium',
+} as const;
+
+type ButtonType = 'button' | 'submit';
+
+type ButtonProperties = {
+    size: (typeof ButtonSize)[keyof typeof ButtonSize];
+    variant: (typeof ButtonVariant)[keyof typeof ButtonVariant];
     label: string;
-    type?: 'button' | 'submit';
+    leftIcon?: ReactNode;
+    rightIcon?: ReactNode;
+    type?: ButtonType;
+    isDisabled?: boolean;
+    className?: string;
+    onClick?: () => void;
+};
+
+const buttonVariantToClasses: Record<
+    (typeof ButtonVariant)[keyof typeof ButtonVariant],
+    string
+> = {
+    [ButtonVariant.PRIMARY]:
+        'rounded-lg bg-lm-yellow-100 text-lm-black-300 hover:bg-lm-yellow-200 disabled:text-lm-grey-200 disabled:bg-lm-grey-300',
+    [ButtonVariant.SECONDARY]:
+        'border border-lm-yellow-100 rounded-lg bg-transparent text-lm-yellow-100 hover:text-lm-yellow-200 hover:border-lm-yellow-200 disabled:text-lm-grey-300 disabled:border-lm-grey-300',
+    [ButtonVariant.TERTIARY]:
+        'bg-transparent text-lm-yellow-100 hover:text-lm-yellow-200 disabled:text-lm-grey-300',
 };
 
-const Button: React.FC<Properties> = ({ type = 'button', label }) => (
-    <button
-        type={type}
-        className="rounded bg-blue-500 px-4 py-2 font-bold text-white hover:bg-blue-700"
-    >
-        {label}
-    </button>
-);
+const Button: React.FC<ButtonProperties> = ({
+    label,
+    variant,
+    size,
+    leftIcon,
+    rightIcon,
+    className,
+    type = 'button',
+    isDisabled = false,
+    ...properties
+}: ButtonProperties): JSX.Element => {
+    const baseClasses =
+        'flex justify-center items-center transition ease-in-out duration-300';
+    const sizeClasses =
+        size === ButtonSize.SMALL
+            ? 'px-4 py-2 h-[32px] text-[14px] font-bold leading-[1.3rem] gap-[4px]'
+            : 'px-6 py-4 h-[54px] text-[16px] font-bold leading-[1.3rem] gap-[8px]';
+    return (
+        <button
+            disabled={isDisabled}
+            type={type}
+            className={getValidClassNames(
+                baseClasses,
+                'w-full',
```

Comment:
I believe it could be included into baseClasses. Base classes could also be inlined